styles_src/default.scss

```diff
@@ -68,51 +67,176 @@
     rgb(250, 250, 250) 0.567324px -1.91785px 0px,
     rgb(250, 250, 250) 1.41734px -1.41108px 0px,
     rgb(250, 250, 250) 1.92034px -0.558831px 0px;
+}
 
+.q-map-code-point {
+  font-family: sans-serif;
+}
+
+%q-map-marker__label--vertical {
   left: 4px;
   transform: translateX(-50%);
+}
+
+%q-map-marker__label--horizontal {
+  line-height: 0.6;
+}
 
+%q-map-marker__label--top {
+  bottom: 1.1em;
   line-height: 1.1;
 }
 
-.q-map-code-point {
-  font-family: sans-serif;
+%q-map-marker__label--bottom {
+  top: 1.1em;
+  line-height: 1.1;
+}
+
+%q-map-marker__label--left {
+  right: 18px;
+}
+
+%q-map-marker__label--right {
+  left: 18px;
 }
 
 .q-map-marker__label--top {
-  bottom: 1.1em;
+  @extend %q-map-marker__label--vertical;
```

Comment:
the `@extend` results in a lot of of css generated.
an alternative could be to separate the logic in the same classes, and then apply both in the markup instead of generating lots of css.